libraries/chain/webassembly/crypto.cpp

```diff
@@ -65,6 +66,50 @@ namespace eosio { namespace chain { namespace webassembly {
       }
    }
 
+   int32_t interface::recover_key_safe( legacy_ptr<const fc::sha256> digest,
+                                        legacy_span<const char> sig,
+                                        legacy_ptr<char> pub,
+                                        legacy_ptr<uint32_t> publen) const {
+      using error_code = eosio::chain::webassembly::error_codes::recover_key_safe;
+
+      try {
+         fc::crypto::signature s;
+         try {
+            datastream<const char*> ds( sig.data(), sig.size() );
+            fc::raw::unpack(ds, s);
+         } catch (...) {
```

Comment:
After some discussion, we would like these `...` catches to be changed to catch fc exception which should be the only exceptions thrown from any of these calls.